### PR TITLE
Biling cli

### PR DESCRIPTION
**Pull request by Andrei-Dolgolev**
mnstr usage generate --month '2024-02' --user-id <user_id> --contracts '{"polygon": [<address_for_events_count>]}'  --output ./reports/customer-2024-02.json

### CODE REVIEW

File: moonstreamapi/moonstreamapi/admin/billing.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+        repeat += 1
+
+        if repeat > max_retries:
+            logger.info("Too many retries")
+            break
+    return data_response.json()
+
+
+def collect_billing(
```
Comment (zomglings) [inline moonstreamapi/moonstreamapi/admin/billing.py:91]: Let us call this `collect_billing_information`. It doesn't actually calculate a dollar value, so `collect_billing` is a bit misleading.

File: moonstreamapi/moonstreamapi/admin/billing.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+) -> Dict[str, Any]:
+    """
+    Collect billing information for a user.
+
+    By user_id or token.
+
+    Collected info:
+
+    Resources:
+        - queries
```
Comment (zomglings) [inline moonstreamapi/moonstreamapi/admin/billing.py:105]: We don't bill by query.

File: moonstreamapi/moonstreamapi/admin/billing.py (hunk 1 above)
Comment (zomglings) [reply moonstreamapi/moonstreamapi/admin/billing.py:91]: Maybe better to call it `collect_usage_information`. For things like number of queries, that's a very important statistic for us to track even though we don't use it for billing.

File: moonstreamapi/moonstreamapi/admin/cli.py [hunk 3]
```diff
@@ -505,6 +519,55 @@ def main() -> None:
         "-n", "--name", required=True, help="Name for the new query"
     )
     create_query_parser.set_defaults(func=queries.create_query_template)
+
+    billing_parser = subcommands.add_parser(
+        "billing", description="Manage Moonstream billing"
```
Comment (zomglings) [inline moonstreamapi/moonstreamapi/admin/cli.py:524]: `"billing"` -> `"usage"`?

File: moonstreamapi/moonstreamapi/admin/cli.py [hunk 4]
```diff
@@ -505,6 +519,55 @@ def main() -> None:
...
+
+    generate_billing_parser.add_argument(
+        "--month",
+        required=True,
+        type=str,
+        help="Month for which to generate billing in YYYY-MM format (e.g. 2021-10)",
+    )
+
+    generate_billing_parser.add_argument(
+        "--user-token",
```
Comment (zomglings) [inline moonstreamapi/moonstreamapi/admin/cli.py:543]: Better if the input is `--user-id`, not `--user-token`.

**Review by zomglings — COMMENTED** [review]
Suggested renaming `billing` to `usage`.

Also suggested `--user-id`, not `--user-token`.

**Comment by Andrei-Dolgolev** [comment]
Block by issues

**Review by kompotkot — APPROVED** [review]
lg

**Pull request merged**